### PR TITLE
Added ConfigureAwait(false) to every awaited async call

### PR DESCRIPTION
**Pull request by Leoltron**
Fixes https://github.com/dnperfors/TestableHttpClient/issues/26 .

### CODE REVIEW

**Comment by dnperfors** [comment]
Thank you very much for opening this pull request, I really appreciate you took the time to at `.ConfigureAwait(false)` for every single async call. However, I don't see a reason why this should be needed inside tests.

As far as I understand, `.ConfigureAwait(false)` should only be added when you are inside a library. The tests are only calling the library itself. A simple console program would normally also not call `.ConfigureAwait(false)`

Also the guidelines clearly state you should use `.ConfigureAwait(false)` in a library itself, but before I do that, I want to have a test that proves that we get a deadlock because we are not using `.ConfigureAwait(false)`.

**Comment by Leoltron** [comment]
It didn’t took much time to do - I made a tool to add `.ConfigureAwait(false)` where async calls were made, and looked for C# repos where such issues were opened to test it. I did find strange that you wanted to add  `.ConfigureAwait(false)` when only tests await asynchronous calls but decided to open PR regardless.

About deadlocks - they cannot happen in your tests (HttpClient is thread-safe and even if tests in the same class share it, no deadlock will ever happen) and there's no asynchronous code in library itself.

In short - yes, there's no reason to add .ConfigureAwait anywhere and this PR and issue can be closed.

**Comment by dnperfors** [comment]
Closing this PR, since it was not the intended change.
